model/fn-execution/src/main/resources/org/apache/beam/model/fnexecution/v1/standard_coders.yaml

```diff
@@ -384,3 +384,31 @@ nested: false
 examples:
   "\x02\x01\x02\x01": {f_bool: True, f_bytes: null}
   "\x02\x00\x00\x04ab\x00c": {f_bool: False, f_bytes: "ab\0c"}
+
+---
+
+# Binary data generated with the python SDK:
+#
+# import typing
+# import apache_beam as beam
+# class Test(typing.NamedTuple):
+#   f_map: typing.Mapping[str,int]
+# schema = beam.typehints.schemas.named_tuple_to_schema(Test)
+# coder = beam.coders.row_coder.RowCoder(schema)
+# print("payload = %s" % schema.SerializeToString())
+# examples = (Test(f_map={}),
+#             Test(f_map={"foo": 9001, "bar": 9223372036854775807}),
+#             Test(f_map={"everything": None, "is": None, "null!": None, "¯\_(ツ)_/¯": None}))
+# for example in examples:
+#   print("example = %s" % coder.encode(example))
+coder:
+  urn: "beam:coder:row:v1"
+  # f_map: map<str, nullable int64>
+  payload: "\n\x15\n\x05f_map\x1a\x0c*\n\n\x02\x10\x07\x12\x04\x08\x01\x10\x04\x12$d8c8f969-14e6-457f-a8b5-62a1aec7f1cd"
+  # map ordering is non-deterministic
+  non_deterministic: True
+nested: false
```

Comment:
If schema coders are always used in a nested context, why is nested: false?

Reply:
Robert was saying that the coders used to encode each attribute value are always nested. In the Java implementation we always call the version of `encode` which does not accept a context (as I understand it most implementations default to the nested encoding in this case):
```
https://github.com/apache/beam/blob/1bf60b99741ff4d0a8b88e93f5241379e1f3962f/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoderGenerator.java#L267-L272
```

Reply:
As it stands, this is confusing for SDK authors writing tests against standard_coders.yaml, as I've got the go testing written I need to explicitly ignore the nested field for the row coders because they're all set to nested:false, rather than nested:true.

This is per my thread on the dev list: https://lists.apache.org/thread.html/r7da098363e6ce607ce96f9fbedb08f9f4757bedd68846aaeba5dd4f0%40%3Cdev.beam.apache.org%3E

Portability only ever supports nested coders. The semantics of standard_coders.yaml say that 
```
#   nested: a boolean meaning whether the coder was used in the nested context. Missing means to
#           test both contexts, a shorthand for when the coder is invariant across context.
```
https://github.com/apache/beam/blob/587dde57cbb2b0095a1fa04b59798d1b62c66f18/model/fn-execution/src/main/resources/org/apache/beam/model/fnexecution/v1/standard_coders.yaml#L24
Meaning that nested: false means that the outer most encoding has the length prefix if necessary.

Structually, there's never a reason for a single schema value to have the wrapped length prefix (it's orthogonal to this aspect of the encoding, as any sub component is always nested as needed), so it's not included in the various payload examples.

So, I re-iterate: Why is nested: false, instead of nested true if the coding is going to be identical in both context?

Reply:
> So, I re-iterate: Why is nested: false, instead of nested true if the coding is going to be identical in both context?

You're not re-iterating. You're asking the same question with a lot of new and different context.

I don't think there's a good reason. If I had to guess - when I added the first row coder test I was new to Beam and coders and I didn't really understand nested vs un-nested, so I just picked one. I'm trying to change it now and it looks like Java is failing some of the nested=true cases - that's probably why. Looking into the failures now.

Reply:
Ack. Thank you. I figured as much. If it's not easy to clean up, I'm fine with the Go coder test blaming "Legacy Java concerns" in a comment for the hackaround.

Reply:
Found the issue in Java. It's a bug in CommonCodersTest that occurs when testing both nested and un-nested. It happens because we share an object for the expected value:
https://github.com/apache/beam/blob/1bf60b99741ff4d0a8b88e93f5241379e1f3962f/runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/CommonCoderTest.java#L193-L196

and it get's mutated when parsing Rows:
https://github.com/apache/beam/blob/1bf60b99741ff4d0a8b88e93f5241379e1f3962f/runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/CommonCoderTest.java#L380

It's a pretty easy fix I can apply here. Is it ok if the row test case just doesn't specify `nested`, or are you thinking it should be `nested: true`?

Reply:
I bet I ran into this way back when and I just opted to pick a (bad) value for `nested` rather than uncovering this issue.

Reply:
Not specifying it would be ideal, as it avoids the hack, so the SDK can only run the "nested" tests, just like the other values.

Reply:
Done! `nested` is no longer specified.